test/analytics/tests/instrumented_events/from_the_server/core_ops_metrics.ts

```diff
@@ -0,0 +1,71 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { Event } from '@kbn/analytics-client';
+import { FtrProviderContext } from '../../../services';
+
+export default function ({ getService }: FtrProviderContext) {
+  const ebtServerHelper = getService('kibana_ebt_server');
+
+  describe('core-ops_metrics', () => {
+    let initialEvent: Event;
+    let secondEvent: Event;
+    let thirdEvent: Event;
+
+    before(async () => {
+      [initialEvent, secondEvent, thirdEvent] = await ebtServerHelper.getLastEvents(3, [
+        'core-ops_metrics',
+      ]);
+    });
+
+    it('should emit the initial ops metrics event', () => {
+      expect(initialEvent.event_type).to.eql('core-ops_metrics');
+      // context assertions
+      expect(initialEvent.context).to.have.property('kibana_uuid');
+      expect(initialEvent.context.kibana_uuid).to.be.a('string');
+      expect(initialEvent.context).to.have.property('pid');
+      expect(initialEvent.context.pid).to.be.a('number');
+
+      // properties assertions. Note: Once we implement schema and collector event validation, we can rely on ts to verify the shape
+      expect(initialEvent.properties).to.have.property('collected_at');
+      expect(initialEvent.properties).to.have.property('process');
+      expect(initialEvent.properties).to.have.property('processes');
+      expect(initialEvent.properties).to.have.property('os');
+      expect(initialEvent.properties).to.have.property('response_times');
+      expect(initialEvent.properties).to.have.property('requests');
+      expect(initialEvent.properties).to.have.property('concurrent_connections');
+    });
+
+    it('should emit the 2nd event with extended context', () => {
+      expect(secondEvent.event_type).to.eql('core-ops_metrics');
+      expect(secondEvent.context).to.have.property('overall_status_level');
+      expect(secondEvent.context).to.have.property('overall_status_summary');
+    });
+
+    it('should emit the 3rd event with updated ops metrics', () => {
+      // we can be fairly certain that at least the os load will have changed after startup.
+      const propertyKeys = ['collected_at', 'os'];
+      propertyKeys.forEach((key: string) => {
+        expect(initialEvent.properties[key]).not.to.be(thirdEvent.properties[key]);
+      });
+    });
+
+    it('should not send a huge json file', () => {
+      const firstEventByteSize = Buffer.byteLength(JSON.stringify(initialEvent));
+      const secondEventByteSize = Buffer.byteLength(JSON.stringify(secondEvent));
+      const thirdEventByteSize = Buffer.byteLength(JSON.stringify(thirdEvent));
+
+      expect(firstEventByteSize).to.be.lessThan(5000);
+      expect(secondEventByteSize).to.be.lessThan(5000);
+      expect(thirdEventByteSize).to.be.lessThan(5000);
+
+      expect(thirdEventByteSize - firstEventByteSize).to.be.greaterThan(0);
```

Comment:
Q: does the third event need to be larger than the first one for any reason?

Reply:
It probably doesn't

Reply:
The third event will have the extended context, while the first has the initial context and _should_ be a slightly larger doc. That being said, ensuring that the _total_ event size isn't bigger than 5kb should be enough to catch any massive increase as the context extends.

Reply:
in that case, the tests for one event type `core-ops_metrics` are relying on another (initially unrelated) piece of data `context`. I'd leave any `context` tests to the context suite. Unless we think this event type should rely on a specific piece of context.